File: src/CENTRO_SALUD/Paciente.java
package CENTRO_SALUD;

import java.util.ArrayList;
import java.util.List;
 
public class Paciente extends Persona {
    private String fechaNacimiento;
    private String sexo;
    private HistoriaClinica historiaClinica; // 1 paciente -> 1 historia clinica
    private List<Cita> citas = new ArrayList<>(); // 1 paciente -> N citas
 
    public Paciente(String idPersona, String dni, String nombres, String apellidos,
                     String fechaNacimiento, String sexo) {
        super(idPersona, dni, nombres, apellidos);
        this.fechaNacimiento = fechaNacimiento;
        this.sexo = sexo;
    }

    public void agregarCita(Cita cita) {
        citas.add(cita);
    }

    public List<Cita> getCitas() { return citas; }
 
    public String getFechaNacimiento() { return fechaNacimiento; }
    public void setFechaNacimiento(String fechaNacimiento) { this.fechaNacimiento = fechaNacimiento; }
 
    public String getSexo() { return sexo; }
    public void setSexo(String sexo) { this.sexo = sexo; }
 
    public HistoriaClinica getHistoriaClinica() { return historiaClinica; }
    public void setHistoriaClinica(HistoriaClinica historiaClinica) { this.historiaClinica = historiaClinica; }
}

File: centro_salud_python/paciente.py
from persona import Persona


class Paciente(Persona):
    """Un paciente del centro de salud. Hereda datos generales de Persona."""

    def __init__(self, id_persona, dni, nombres, apellidos, fecha_nacimiento, sexo):
        super().__init__(id_persona, dni, nombres, apellidos)
        self._fecha_nacimiento = fecha_nacimiento
        self._sexo = sexo
        self._historia_clinica = None  # 1 paciente -> 1 historia clinica
        self._citas = []               # 1 paciente -> N citas

    def agregar_cita(self, cita):
        self._citas.append(cita)

    def get_citas(self):
        return self._citas

    def get_fecha_nacimiento(self):
        return self._fecha_nacimiento

    def set_fecha_nacimiento(self, fecha_nacimiento):
        self._fecha_nacimiento = fecha_nacimiento

    def get_sexo(self):
        return self._sexo

    def set_sexo(self, sexo):
        self._sexo = sexo

    def get_historia_clinica(self):
        return self._historia_clinica

    def set_historia_clinica(self, historia_clinica):
        self._historia_clinica = historia_clinica
